smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/CodegenVisitor.java

```diff
@@ -84,8 +84,8 @@
     CodegenVisitor(PluginContext context) {
         settings = TypeScriptSettings.from(context.getModel(), context.getSettings());
         nonTraits = context.getModelWithoutTraitShapes();
-        model = context.getModel();
-        service = settings.getService(model);
+        Model baseModel = context.getModel();
+        service = settings.getService(baseModel);
```

Comment:
Should this be set after `preprocessModel` is run?

Reply:
I see. All the shapes with reference to `model` should be defined after `preprocessModel` is applied. Rigth?